src/runtime/virtcontainers/clh.go

```diff
@@ -311,10 +305,10 @@ func (clh *cloudHypervisor) createSandbox(ctx context.Context, id string, networ
 	// Overwrite the default value of HTTP API socket path for cloud hypervisor
 	apiSocketPath, err := clh.apiSocketPath(id)
 	if err != nil {
-		clh.Logger().Info("Invalid api socket path for cloud-hypervisor")
-		return nil
+		clh.Logger().WithError(err).Info("Invalid api socket path for cloud-hypervisor")
+	} else {
+		clh.state.apiSocket = apiSocketPath
```

Comment:
@liubin Would you please elaborate why we don't want to abort  `createSandbox()` right away when function `apiSocketPath` failed? Thank you.

I may miss some context. But two reasons that I can see why we decided to abort immediately: 1) `apiSocketPath ` will return `""` on error, and it won't be a valid API socket for kata runtime to control cloud-hypervisor; 2) same scenario for qemu, we aborted `createSandbox` right away when `createQmpSocket()` fails [here](https://github.com/kata-containers/kata-containers/blob/1c222c75ac046ed0105fac1e00f7defcbd35bad7/src/runtime/virtcontainers/qemu.go#L558-L561).

Reply:
Hi @likebreath, the original code is `return nil` if `apiSocketPath` failed and the default HTTP API socket will be used(only my guess).

And lines after 310, include [virtiofsd](https://github.com/kata-containers/kata-containers/commit/a2d3f9f32de6d95094dcb06c505a33f7fd8bc531#) and [SGXEPCSize](https://github.com/kata-containers/kata-containers/commit/6df165c19d7209571a229e8826e97d33a5aa7a24#) are committed after the original version.

So I think if the `apiSocketPath`  failed, the code should not return immediately, but to continue to config virtiofs. 

If this is not reasonable, maybe should the `return nil` be replaced by `return err` if `clh.apiSocketPath` failed.

Reply:
@liubin Right.. I mis-read the code (thought we were aborting by returning an error). I think that is actually a bug: "the original code is return nil if apiSocketPath failed". And I revisited the code and I think we don't have  a `default` value for CLH API socket..

So it looks more reasonable to me with your proposal in the above comment: replace `return nil` with `return err`. Please let me know if you have other thoughts. 

Thanks again for promoting the issue and fixing it.

Reply:
@liubin I agree with @likebreath , I feel better to return an `err` instead of continue .

Reply:
hi @jcvenegas @likebreath , I will fix it by `return err` as discussed above.